core/messagechains/src/lib.rs

```diff
@@ -0,0 +1,583 @@
+use std::collections::HashMap;
+use std::collections::VecDeque;
+use std::fmt::Debug;
+use std::hash::Hash;
+
+use log;
+
+pub mod sha256;
+pub mod wasm_wrapper;
+
+/// Generic device ID
+pub trait DeviceId: Sized + Hash + Eq + Ord + Clone + Debug + AsRef<[u8]> {}
```

Comment:
Is there ever a case (in the foreseeable future) where we'd want/need the the device ids to be anything but some specific type (either a String or a `Vec<u8>` or a `[u8; 32]` or something)?